### ISSUE TITLE
keyword-{false,true}-as-identifier.rs tests broken

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust#33010]
In the process of porting compiletest to use JSON, @jntrnr and I discovered that these two tests were broken. The older test runner had some bugs that let them squeak through:

```rust
fn main() {
    let false = "foo"; //~ error: ident
}
```

yields:

```
keyword-false-as-identifier.rs:14:9: 14:14 error: mismatched types:
 expected `&str`,
    found `bool`
(expected &-ptr,
    found bool) [E0308]
keyword-false-as-identifier.rs:14     let false = "foo"; //~ error: ident
                                          ^~~~~
keyword-false-as-identifier.rs:14:9: 14:14 help: run `rustc --explain E0308` to see a detailed explanation
error: aborting due to previous error
```

**Comment by nikomatsakis** [comment]
cc @nrc maybe related to parser identifier changes?

**Comment by petrochenkov** [comment]
`false` is a valid literal pattern. Something like
```
    let 1 = "foo";
    let "a" = "foo";
```
isn't caught during parsing as well.

**Comment by jonas-schievink** [comment]
Well, technically `false` *is* a pattern (a refutable one, so you always get `error: refutable pattern in local binding: `true` not covered`). What's the last version that rejected this with the message from the test?

**Comment by nikomatsakis** [comment]
@jonas-schievink ah, good point -- perhaps it was indeed a bug fix :)

**Comment by steveklabnik** [comment]
Closing as "not actually a bug", then. :smile:
